### PR TITLE
FIX: make sure "opener window" is not "current window" during onmiauth callback

### PR DESCRIPTION
**Pull request by xrav3nz**
More details: https://meta.discourse.org/t/oauth2-basic-support/33879/168?u=xrav3nz

### CODE REVIEW

**Comment by davidtaylorhq** [comment]
Did you manage to reproduce the case where `window.opener === window`? It seems super weird that that would happen - maybe a browser bug?

In any case, this change looks pretty safe to me :)

**Comment by xrav3nz** [comment]
I can only repro on the site that's experiencing this (with different browsers), but not locally. 🤕 It's indeed super weird.

File: app/assets/javascripts/omniauth-complete.js.no-module.es6 [hunk 1]
```diff
@@ -4,7 +4,7 @@
   ).dataset;
   const parsedAuthResult = JSON.parse(authResult);
 
-  if (!window.opener) {
+  if (!window.opener || window.opener === window) {
```
Comment (SamSaffron) [inline app/assets/javascripts/omniauth-complete.js.no-module.es6]: Maybe we go with the safer?
```suggestion
  if (!window.opener || !window.opener.Discourse || !window.opener.Discourse.authenticationComplete) {
```

**Comment by xrav3nz** [comment]
Reapplied the changes manually because of `prettier` 😅

**Comment by SamSaffron** [comment]
Looks good to me! thanks

**Pull request merged**